packages/server/src/models/user-shelf.entity.ts

```diff
@@ -0,0 +1,41 @@
+import { Column, Entity, PrimaryColumn } from 'typeorm';
+import { UserShelfRoles, UserShelfActions } from '../constants/user-shelf';
+
+abstract class ShelfDuties {
+  @PrimaryColumn()
+  shelfId: number;
+
+  @Column('simple-enum', { enum: UserShelfRoles })
+  role: UserShelfRoles;
+  default: UserShelfRoles.KEEPER;
+
+  @Column('simple-enum', { enum: UserShelfActions, array: true })
+  actions: UserShelfActions[];
+}
+
+@Entity()
+export class UserShelfDuties extends ShelfDuties {
+  @PrimaryColumn()
+  userId: string;
+}
+
+@Entity()
+export class UserShelfInvitation extends ShelfDuties {
+  @PrimaryColumn()
+  token: string;
```

Comment:
I would suggest to rename it as 'invitationToken'

Reply:
I started with `invitationToken` to stay in sync with other entities, but then I thought `token` is enough, cause there are no other tokens in this table, and it's not an external key or somethings.

Reply:
I don't insist, guys, please share your thoughts on this. I'll do as more comfortable for the team.

Reply:
I like this naming more, i.e. to me naming like `token` is a bit confusing here since it has a field `userEmail`

Reply:
Fixed